lightning/src/onion_message/messenger.rs

```diff
@@ -1053,20 +1100,49 @@ where
 		)
 	}
 
-	fn handle_onion_message_response<T: OnionMessageContents>(
+	/// Handles the response to an [`OnionMessage`] based on its [`ResponseInstruction`],
+	/// enqueueing any response for sending.
+	///
+	/// This function is useful for asynchronous handling of [`OnionMessage`]s.
+	///
+	/// Handlers have the option to return [`ResponseInstruction::NoResponse`], indicating that
+	/// no immediate response should be sent. Instead, they can transfer the associated [`Responder`]
+	/// to another task responsible for generating the response asynchronously. Subsequently, when
+	/// the response is prepared and ready for sending, that task can invoke this method to enqueue
+	/// the response for delivery.
```

Comment:
Let's combine these paragraphs as the second is describing asynchronous handling.

Reply:
Done! [d8adbb0](https://github.com/lightningdevkit/rust-lightning/commit/d8adbb0a5b7dcc39a41596274129ff3d69ec08de)
Thanks!